### PR TITLE
XAPTC-69  Consistent ValidateHeaderAndName function in Golang

### PR DESCRIPTION
**Pull request by frankquinn**
Changed all remaining instances of "ValidateHeaderAndName" to "if !ValidateHeaderAndName{ return}" for consistency. There was only one instance that was left to be changed.

### CODE REVIEW

**Comment by ConorSmyth** [comment]
What about these, we are meant to always check to see if the header is valid for these too and they should be consistent. 

- postConfigs
- getAllTestCases
- getAllTestSuites

Since you are already making changes for this function. It is a good time to address this function as well. Refactor the function ValidateHeaderAndName as it is doing 2 things at once and is particular bad programming practice.

**Comment by jenny-flynn** [comment]
Agree, @ConorSmyth That function should just return an error that can be caught and handled in the main function. 

e.g. 

`if !ValidateFileNameAndHeader(testSuitePathDir, testSuiteName){
    rw.WriteHeader(http.StatusBadRequest)
}
`

And the function can be refactored:
`func ValidateFileNameAndHeader(header, name string) bool {

	if len(name) < 1 {
		logrus.Error("File Name is Empty")
		return false
	}

	if len(header) <= 1 {
		logrus.Error("No Header Path Found")
		return false
	}
	return true
}
`

**Comment by CameliaFarcas** [comment]
@jenny-flynn , @ConorSmyth , @AlexeiSuleac please review the split function ValidateNameAndHeader

File: uiServices/src/configServices.go [hunk 1]
```diff
@@ -193,3 +199,19 @@ func putConfigs(rw http.ResponseWriter, req *http.Request) {
...
 	rw.WriteHeader(http.StatusNoContent)
 }
+
+func putConfigFileName(rw http.ResponseWriter, req *http.Request) {
+	path := getConfigHeader(req)
+	configFileName := chi.URLParam(req, "configFileName")
+	newConfigFileName := chi.URLParam(req, "newConfigFileName")
+
+	if len(newConfigFileName) < 1 {
+		logrus.Error("File path does not exist")
```
Comment (ConorSmyth) [inline uiServices/src/configServices.go]: You can use this function here IsHeaderValid

File: uiServices/src/configServices.go [hunk 2]
```diff
@@ -28,15 +28,21 @@ func getConfigHeader(req *http.Request) string {
 }
 
 func ValidateFileNameAndHeader(rw http.ResponseWriter, req *http.Request, header, name string) bool {
+	return IsNameValid(name, rw) && IsHeaderValid(header, rw)
```
Comment (jenny-flynn) [inline uiServices/src/configServices.go]: see my comment from yesterday about removing the responsewriter functionality also

File: uiServices/src/configServices.go (hunk 1 above)
Comment (jenny-flynn) [inline uiServices/src/configServices.go]: log message is incorrect.
1. is it file path or file name?
2. message should say it is empty

File: uiServices/src/configServices.go [hunk 3]
```diff
@@ -193,3 +199,19 @@ func putConfigs(rw http.ResponseWriter, req *http.Request) {
...
+	configFileName := chi.URLParam(req, "configFileName")
+	newConfigFileName := chi.URLParam(req, "newConfigFileName")
+
+	if len(newConfigFileName) < 1 {
+		logrus.Error("File path does not exist")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	os.Rename(fmt.Sprintf("%s%s.xml", path, configFileName), fmt.Sprintf("%s%s.xml", path, newConfigFileName))
```
Comment (jenny-flynn) [inline uiServices/src/configServices.go]: check for a returned error, and return 500 if there is an error

File: uiServices/src/configServices.go [hunk 4]
```diff
@@ -205,13 +214,21 @@ func putConfigFileName(rw http.ResponseWriter, req *http.Request) {
...
 		return
 	}
 
-	os.Rename(fmt.Sprintf("%s%s.xml", path, configFileName), fmt.Sprintf("%s%s.xml", path, newConfigFileName))
+	configFilePath := fmt.Sprintf("%s%s.xml", path, configFileName)
+	newConfigFilePath := fmt.Sprintf("%s%s.xml", path, newConfigFileName)
 
+	err := os.Rename(configFilePath, newConfigFilePath)
+	if err != nil {
+		logrus.Println("File was not updated", err)
```
Comment (jenny-flynn) [inline uiServices/src/configServices.go]: logrus.Error

File: uiServices/src/testCaseServices.go [hunk 5]
```diff
@@ -128,12 +129,17 @@ func putTestCase(rw http.ResponseWriter, req *http.Request) {
...
 	testCasePathDir := getTestCaseHeader(req)
-	if len(testCasePathDir) <= 1 {
-		logrus.Error("No file directory entered")
-		rw.WriteHeader(http.StatusBadRequest)
+
+	if !IsPathDirValid(testCasePathDir, rw) {
 		return
 	}
 
+	// if len(testCasePathDir) <= 1 {
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: remove commented-out code

**Comment by CameliaFarcas** [comment]
@jenny-Flynn, here is the func ValidateFileNameAndHeader changed, based on your feedback

**In configServices.go we split the function in two**
func IsHeaderValid(header string) error {
	if len(header) <= 1 {
		logrus.Error("No Header Path Found")
		return new error()
	}
	return nil
}

func IsNameValid(name string) error {
	if len(name) < 1 {
		logrus.Error("File Name is Empty")
	}
	return nil
}

**Comment by CameliaFarcas** [comment]
@jenny-flynn  here is the func ValidateFileNameAndHeader, based on your feedback

_In configServices.go the function is split in two:_
func IsHeaderValid(header string) error {
	if len(header) <= 1 {
		logrus.Error("No Header Path Found")
		return fmt.Errorf("No Header Path Found")
	}
	return nil
}

func IsNameValid(name string) error {
	if len(name) < 1 {
		logrus.Error("File Name is Empty")
		return fmt.Errorf("File Name is Empty")
	}
	return nil
}

File: uiServices/src/testCaseServices.go (hunk 5 above)
Comment (CameliaFarcas) [reply uiServices/src/testCaseServices.go]: Leave a commen@jenny-flynn  here is the func ValidateFileNameAndHeader, based on your feedback

**_In configServices.go the function is split in two:_**
```
func IsHeaderValid(header string) error {
	if len(header) <= 1 {
		logrus.Error("No Header Path Found")
		return fmt.Errorf("No Header Path Found")
	}
	return nil
}

func IsNameValid(name string) error {
	if len(name) < 1 {
		logrus.Error("File Name is Empty")
		return fmt.Errorf("File Name is Empty")
	}
	return nil
}

```
**and in the main function**:

```
if err := IsHeaderValid(testSuitePathDir); err != nil {
		rw.WriteHeader(http.StatusBadRequest)
		return
	}

	if  err:= IsNameValid(testSuiteName); err != nil {
		rw.WriteHeader(http.StatusBadRequest)
		return
	}
```

File: uiServices/src/configServices.go [hunk 6]
```diff
@@ -112,7 +121,13 @@ func getConfigs(rw http.ResponseWriter, req *http.Request) {
 	configPathDir := getConfigHeader(req)
 	configName := chi.URLParam(req, "configName")
 
-	if !ValidateFileNameAndHeader(rw, req, configPathDir, configName) {
+	if err := IsHeaderValid(configPathDir); err != nil {
```
Comment (jenny-flynn) [inline uiServices/src/configServices.go]: nice

**Pull request merged**